project_resource_calendar/models/calendar_event.py

```diff
@@ -161,27 +160,27 @@ def _check_resources_double_book(self):
             )
             if not any(room) and not any(equipment):
                 continue
-            overlaps = self.env['calendar.event'].search([
+            events = self.env['calendar.event'].search([
                 ('id', '!=', record.id),
-                ('start', '<', record.stop),
-                ('stop', '>', record.start),
             ])
-            for resource in overlaps.mapped(lambda s: s.room_id):
-                if resource.id == record.room_id.id:
-                    raise ValidationError(
-                        _(
-                            'The room, %s,  cannot be double-booked '
-                            'with any overlapping meetings or events.',
-                        ) % resource.name,
-                    )
-            for resource in overlaps.mapped(lambda s: s.equipment_ids):
-                if resource.id in record.equipment_ids.ids:
-                    raise ValidationError(
-                        _(
-                            'The resource, %s, cannot be double-booked '
-                            'with any overlapping meetings or events.',
-                        ) % resource.name,
-                    )
+            for event in events:
+                if (event.start < record.stop) & (event.stop > record.start):
```

Comment:
The logic is good. but maybe if you create a method:
def record_intercets_event(self, record, event):
    if (event.start < record.stop) & (event.stop > record.start):
        return True
    else:
        return False

and call this method in place, it's more readable, don't you think?